ibis/backends/clickhouse/registry.py

```diff
@@ -74,6 +74,63 @@ def formatter(translator, expr):
     return formatter
 
 
+def _array_index_op(translator, expr):
+    op = expr.op()
+
+    arr = op.args[0]
+    idx = op.args[1]
+
+    arr_ = translator.translate(arr)
+    idx_ = _parenthesize(translator, idx)
+
+    correct_idx = f'if({idx_} >= 0, {idx_} + 1, {idx_})'
+
+    return f'arrayElement({arr_}, {correct_idx})'
+
+
+def _array_repeat_op(translator, expr):
+    op = expr.op()
+    arr, times = op.args
+
+    arr_ = _parenthesize(translator, arr)
+    times_ = _parenthesize(translator, times)
+
+    return f'''
+        (select arrayFlatten(groupArray(arr)) from
+        (
+            select {arr_} as arr
+            from system.numbers
+            limit {times_}
+        ))
+    '''
+
+
+def _array_slice_op(translator, expr):
+    op = expr.op()
+    arg, start, stop = op.args
+
+    start_ = _parenthesize(translator, start)
+    arg_ = translator.translate(arg)
+
+    start_correct_ = f'if({start_} < 0, {start_}, {start_} + 1)'
```

Comment:
What are the semantics when `start_` is negative?

Reply:
When `start_` is negative, `arraySlice` works as a python slice with only the "start" argument, for example, if you define `arr[-2:]` in python, then for CH it will be `arraySlice(arr, -2)`.

Reply:
Got it, thanks.